service/debugger/debugger.go

```diff
@@ -1475,6 +1487,158 @@ func (d *Debugger) Functions(filter string) ([]string, error) {
 	return funcs, nil
 }
 
+type TraceFunc struct {
+	// Function to be traced
+	Func     *proc.Function
+	// To keep track of depth in callgraph relative to root function
+	Depth    int
+	// To guard against adding duplicates to list of functions to be traced
+	visited  bool
+	// To avoid disassembling function multiple times
+	dis      bool
+	// To quickly access child functions
+	children []*proc.Function
+}
+type TraceFuncptr *TraceFunc
+
+var TraceMap = make(map[string]TraceFuncptr)
+
+func filter(fname string) bool {
+	idx := strings.Index(fname, "runtime.")
+	idx2 := strings.Index(fname, "runtime.defer")
+	idx3 := strings.Index(fname, "runtime.gorecover")
+	idx4 := strings.Index(fname, "runtime.gopanic") 
+	// Avoid runtime/internal functions to figure in the trace function list as they dont have a stack associated
+	idx5 := strings.Index(fname, "runtime/internal")
+	if (idx != -1 || idx5 != -1) && idx2 == -1 && idx3 == -1 && idx4 == -1 {
+		// Except certain special functions do not traverse runtime.* functions
+		return false
+	}
+	return true
+}
+
+func dequeue(queue *list.List) TraceFuncptr {
+	element := queue.Front()
+	queue.Remove(element)
+	data := element.Value
+	F, ok := data.(TraceFuncptr)
+	if !ok {
+		fmt.Printf("Type error\n")
+		os.Exit(1)
+	}
+	return F
+}
+
+func traverse(t proc.ValidTargets, f *proc.Function, depth int, FollowCalls int) ([]string, error) {
+
+	var queue *list.List
+	queue = list.New()
+	queue.Init()
+
+	funcs := []string{}
+
+	rootnode := &TraceFunc{Func: new(proc.Function), Depth: depth, children: []*proc.Function{}, visited: false, dis: false}
+	rootnode.Func = f
+
+	TraceMap[f.Name] = rootnode
+	queue.PushBack(TraceMap[f.Name])
```

Comment:
If we go with a slice instead, you could do:

```
queue := make([]traceFuncPtr, 0, 20)
queue[0] = rootnode
```

Reply:
done